ambari-server/src/main/java/org/apache/ambari/server/api/query/render/ClusterBlueprintRenderer.java

```diff
@@ -299,34 +323,36 @@ private Resource createBlueprintResource(TreeNode<Resource> clusterNode) {
         Map<String, Object> ServiceComponentInfoMap = component.getPropertiesMap().get("ServiceComponentInfo");
 
         if (ServiceComponentInfoMap.get("recovery_enabled").equals("true")) {
-          globalRecoveryEnabled = true;
-          property.put("name", ServiceInfoMap.get("service_name").toString());
-          property.put("recovery_enabled", "true");
+          serviceProperty.put("name", ServiceInfoMap.get("service_name").toString());
+          serviceProperty.put("recovery_enabled", "true");
 
           //component_settings population
-          componentProperty = new HashMap<>();
+          HashMap<String, String> componentProperty = new HashMap<>();
           componentProperty.put("name", ServiceComponentInfoMap.get("component_name").toString());
           componentProperty.put("recovery_enabled", "true");
+          componentSettingValue.add(componentProperty);
         }
       }
 
-      if (!property.isEmpty())
-        serviceSettingValue.add(property);
-      if (!componentProperty.isEmpty())
-        componentSettingValue.add(componentProperty);
+      if (!serviceProperty.isEmpty())
+        serviceSettingValue.add(serviceProperty);
     }
-    //recovery_settings population
-    property = new HashMap<>();
-    if (globalRecoveryEnabled) {
-      property.put("recovery_enabled", "true");
-    } else {
-      property.put("recovery_enabled", "false");
+
+    // Add cluster settings
+    Set<Map<String, String>> clusterSettings = new HashSet<>();
+    TreeNode<Resource> settingsNode = clusterNode.getChild("settings");
+    if (null != settingsNode) {
+      for (TreeNode<Resource> clusterSettingNode: settingsNode.getChildren()) {
+        Map<String, Object> nodeProperties = clusterSettingNode.getObject().getPropertiesMap().get("ClusterSettingInfo");
+        String key = Objects.toString(nodeProperties.get("cluster_setting_name"));
+        String value = Objects.toString(nodeProperties.get("cluster_setting_value"));
```

Comment:
Should be reusing constants from `ClusterSettingResourceProvider`.

Reply:
fixed